cpp/src/lance/io/project.cc

```diff
@@ -60,6 +68,18 @@ ::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> Project::Execute(
       return result.status();
     }
     auto [indices, values] = result.ValueUnsafe();
+    assert(indices->length() == values->num_rows());
+    if (limit_) {
+      auto offset_and_len = limit_->Apply(indices->length());
```

Comment:
Is there no macro to simplify this Optional.Map operation?

Reply:
https://en.cppreference.com/w/cpp/utility/optional  seems not applicable before c++23?